### ISSUE TITLE
Why fingerprint is same in private mode as well  ?

### ISSUE THREAD

**Issue by prk2331** [fingerprintjs/fingerprintjs#861]
Team can you please give me a knowledge that why this is same in incognito mode as well, as last month I run the Sitecore Product there I saw the browser Id always different in incognito mode that time. and I am applying Fingerprint concept in my current project to identify anonymous guest, But while seeing this type of changing behavior in incognito mode I am little bit confused in both concepts.
Can anyone here who can guide me on this

Thanks

**Comment by prk2331** [comment]
In chrome + chrome incognito    Fingerprint is not changing.
 but as I saw in In firefox + firefox incognito is changing????

why this type of inconsitency ? is happening

**Comment by JuroUhlar** [comment]
Hi @prk2331! 

FingerprintJS takes a large number of attributes from your browser (features, fonts, plugins, hardware parameters, etc., [see the demo](https://fingerprintjs.github.io/fingerprintjs/) to get an idea) a hashes them into a single `visitorId` string. 

These signals are the same in incognito mode as they are in normal mode, therefore FingerprintJS gives you the same `visitorId` (I just tested it in both Chrome and Firefox). Which version of Firefox are you using? 

If you are trying to identify if a visitor is using incognito mode, FingerprintJS (the open-source library) does not have a field for this in the result, but [Fingerprint Pro does](https://fingerprint.com/). Developer plans are free for up to 20k identification requests per month.

**Comment by makma** [comment]
Hey @prk2331 ,
In general, we try to use only signals that are stable between normal and incognito modes. This results in the same identifier. Having said that, in some browsers/systems, this behavior might not be achieved just using the open source FingerprintJS and this feature is offered as is, without any guarantees.

**Comment by JuroUhlar** [comment]
Hey @chitralverma I have tested Privacy Badger on the FingerprintJS demo here: https://fingerprintjs.github.io/fingerprintjs/ 
And got the same result in both normal and incognito mode (same for Fingerprint Pro). Privacy Badger seems to focus on third-party tracking, FingerprintJS allows you to identify visitors on your own site in a first-party context, there are no cookies to delete or requests to block.

**Comment by JuroUhlar** [comment]
Hi @ezreall! 

> Why is the identifier on the same device and the same chrome platform?  

Can you rephrase the question, please? I am not sure I understand 

> Just change the Gmail account and the identifier will change 

Do you mean changing the Google profile signed in to the Chrome window? It can change the identifier but doesn't have to. For example, if the profiles have different preferred languages, they will have different identifiers. But if the profile settings are the same, they will have the same identifier. You can transparently check this behavior: 

* open https://fingerprintjs.github.io/fingerprintjs/ in both profiles
* copy the two sets of _Entropy components_ into https://www.diffchecker.com/text-compare/. 

You will see which attributes differ between the two Chrome profiles.

>  I also get another error in one case: gest mode chrome and edge default are generated together. same identifier 

Edge is using Chromium as a rendering engine under the hood, it is therefore expected that their browser fingerprints are the same (provided they have the same language settings, etc.). You could consider [adding your own entropy components](https://github.com/fingerprintjs/fingerprintjs/blob/master/docs/extending.md) (like a user agent) to differentiate them. 

FingerprintJS uses a simple and transparent hashing algorithm to generate the browser identifier. You can add or exclude individual entropy components according to your needs. But if its limitations are a problem for your use case, consider [upgrading to Fingerprint Identification](https://fingerprint.com/github/) which has a much more advanced and accurate approach to browser identification built on top of FingerprintJS.

**Comment by Finesse** [comment]
Closing this because it's a question, not an issue, and it has been answered
